spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindArgumentAssertions.java

```diff
@@ -0,0 +1,146 @@
+package edu.umd.cs.findbugs.detect;
+
+import org.apache.bcel.Const;
+import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.classfile.Method;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.OpcodeStack;
+import edu.umd.cs.findbugs.OpcodeStack.Item;
+import edu.umd.cs.findbugs.ba.ClassContext;
+import edu.umd.cs.findbugs.ba.XMethod;
+
+/**
+ * This detector can find Assertions that try to validate method
+ * arguments.
+ */
+public class FindArgumentAssertions extends AbstractAssertDetector {
+
+    public FindArgumentAssertions(BugReporter bugReporter) {
+        super(bugReporter);
+    }
+
+    /**
+     * Only interested in public methods
+     */
+    @Override
+    public void visit(Method obj) {
+        if (!obj.isPublic())
+            return;
+    }
+
+    /**
+     * Only interested in public classes
+     */
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        JavaClass ctx = classContext.getJavaClass();
+        // Break out of analyzing this class if not public
+        if (!ctx.isPublic())
+            return;
+        super.visitClassContext(classContext);
+    }
+
+    /**
+     * Checks if the methods paramaeter is an initial arg.
+     */
+    private boolean isInitialArg() {
+        XMethod m = getXMethodOperand();
+        int numPar = m.getNumParams();
+        // Get values from the stack
+        for (int i = 0; i < numPar; i++) {
+            Item item = stack.getStackItem(i);
+            if (item.isInitialParameter())
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if the opcode is a method invocation false otherwise
+     */
+    private boolean isMethodCall(int seen) {
+        boolean methodCall = false;
```

Comment:
This could be inlined instead,

    return seen == Const.INVOKESTATIC
        || seen == Const.INVOKEVIRTUAL
        ...

Reply:
I think this one can be resolved, since it got modified in fce1630f06bd8ab5a8705b2cce6506ed3f6fd6e3.